src/__experimental__/components/decimal/decimal.stories.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import {
+  boolean, text, number, select
+} from '@storybook/addon-knobs';
+import { action } from '@storybook/addon-actions';
+import { State, Store } from '@sambego/storybook-state';
+import Decimal from './decimal.component';
+import OptionsHelper from '../../../utils/helpers/options-helper';
+import notes from './documentation';
+
+const store = new Store({
+  value: Decimal.defaultProps.value
+});
+
+const setValue = (evt) => {
+  action('onChange')(evt);
+  store.set({ value: evt.target.value });
+};
+
+storiesOf('Experimental/Decimal', module)
+  .addParameters({
+    info: {
+      propTablesExclude: [State]
+    }
+  }).add('default', () => {
+    const align = select(
+      'align',
+      OptionsHelper.alignBinary,
+      Decimal.defaultProps.align
+    );
+    const precision = number('precision', Decimal.defaultProps.precision);
+    const inputWidth = number('inputWidth', Decimal.defaultProps.inputWidth);
+    const fieldHelp = text('fieldHelp', 'This text provides help for the input.');
+    const label = text('label', 'Example Decimal');
+    const labelInline = label ? boolean('labelInline', false) : undefined;
+    const labelWidth = labelInline ? number('labelWidth', 30) : undefined;
+    const labelAlign = labelInline ? select(
+      'labelAlign',
+      OptionsHelper.alignBinary,
+      OptionsHelper.alignBinary[0]
+    ) : undefined;
+    const labelHelp = label ? text('labelHelp', 'This text provides help for the label.') : undefined;
+
+    return (
+      <State store={ store }>
+        <Decimal
+          align={ align }
+          precision={ precision }
+          inputWidth={ inputWidth }
+          fieldHelp={ fieldHelp }
+          label={ label }
+          labelInline={ labelInline }
+          labelWidth={ labelWidth }
+          labelAlign={ labelAlign }
+          labelHelp={ labelHelp }
```

Comment:
you could just import `Textbox` props from the `Textbox` story:
`{ ...getTextboxStoryProps() }`

Reply:
I'm not sure I want to do I? Those variables are used to enable the Storybook knobs...

Maybe I'm misunderstanding?

Reply:
inputWidth, fieldHelp and label knobs are already specified in the Textbox story, so instead of repeating it we could reuse them.

Reply:
an example:
https://github.com/Sage/carbon/blob/master/src/__experimental__/components/number/number.stories.js#L35

Reply:
I'm not sure this is going to work for `Decimal` without re-writing your `getTextboxStoryProps` method. I include some defaults for these properties in `Decimal`, and my `inputWidth` is not reliant on he `labelInline` being defined.

In fact, looking at what you've done in `getTextboxStoryProps`, I think that the definitions for `labelWidth` and `inputWidth` are at odds with some of the styling work I've done in this PR. I've updated so that the `inputWidth` gets defined as a percentage of the remaining width (total width - `labelWidth`). Check out: https://github.com/Sage/carbon/pull/1970/files#diff-1ae771b61db1ee50897f7001bdc23d1dR22

Reply:
`inputWidth` prop should cause the same effects in all `Textbox` based components, if it should not be reliant on the `labelInline` then it should be a standalone prop in the `Textbox` also. 

I still think that you should reuse these `Textbox` props.

Reply:
`labelWidth` and `inputWith` should behave the same for all input components, and we should really try and replicate the current behaviour to mitigate any issues during upgrade.